src/components/container/IntradayHeartRateChart/IntradayHeartRateChart.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useContext } from 'react'
+import { DateRangeContext } from '../../presentational/DateRangeCoordinator/DateRangeCoordinator'
+import { add, startOfDay } from 'date-fns'
+import { LayoutContext, LoadingIndicator } from '../../presentational'
+import { HalfDayData, FullDayData } from "./IntradayHeartRateChart.data"
+import { ColorDefinition, resolveColor } from '../../../helpers/colors'
+import { useInitializeView } from '../../../helpers/Initialization'
+import { DeviceDataV2Aggregate, DeviceDataV2AggregateQuery, DeviceDataV2Namespace } from '@careevolution/mydatahelps-js'
+import { hrIntradayDataProvider } from '../../../helpers/heart-rate-data-providers'
+import LineChart, { LineChartAxis, LineChartDataPoint } from '../../presentational/LineChart/LineChart'
+
+export type AggregationOption = "avg" | "count" | "min" | "max" | "sum";
+export type IntradayHeartRatePreviewState = "Default" | "CompleteDataWithThresholds" | "PartialDataWithThresholds" | "NoData";
+
+export interface IntradayChartThreshold {
+    value: number
+    referenceLineColor?: ColorDefinition
+    overThresholdLineColor?: ColorDefinition
+}
+
+export interface IntradayHeartRateChartProps {
+    previewState?: IntradayHeartRatePreviewState,
+    dataSource: DeviceDataV2Namespace,
+    aggregationOption: AggregationOption,
+    aggregationIntervalAmount: number,
```

Comment:
Let's just call this "aggregationIntervalMinutes" and not have aggregationIntervalType...this is intraday so no need to complicate with extra params